### ISSUE TITLE
Not able to inject a custom schema service

### ISSUE THREAD

**Issue by mloureiro** [Youshido/GraphQLBundle#47]
tl;dr

Can't Change or initialize the `graphql.schema` service even through compiler pass.

PR: https://github.com/Youshido/GraphQLBundle/pull/48

Hi 

First of all, thanks for all the work in GraphQL and GraphQLBundle.

I'm currently trying to inject the Fields into the Schema.
So for that, I can't simply rely on the config `graphql.schema_class` since the initialization is also needed.
Because of that I've attempted to override the `graphql.schema` service, and due to the way Symfony works, when there is a duplicated service it ignores/picks one, so since the service is already defined in this [`services.xml`](https://github.com/Youshido/GraphQLBundle/blob/5d17ce4ef7064e875bcf4e07d84ffd4b9859b2b2//Resources/config/services.xml#L14) so a compiler pass is needed.

Then I've tried simply to make a compiler pass with:
```php
$container->set('graphql.schema', $container->get('acme.graphql.schema'));
```

Although even if the service is initialized, [there is a class check](https://github.com/Youshido/GraphQLBundle/blob/5d17ce4ef7064e875bcf4e07d84ffd4b9859b2b2/Controller/GraphQLController.php#L42) that will fail since the service is initialized without the class name and [initializes the class anyway](https://github.com/Youshido/GraphQLBundle/blob/5d17ce4ef7064e875bcf4e07d84ffd4b9859b2b2/Controller/GraphQLController.php#L45)

Bonus, even if we define the service, we [MUST define the schema class](https://github.com/Youshido/GraphQLBundle/blob/5d17ce4ef7064e875bcf4e07d84ffd4b9859b2b2//Controller/GraphQLController.php#L37) which is not relevant.

I am not a Symfony expert, if I'm missing or doing something wrong please tell me.

I'll make a PR for this issue

**Comment by MLoureiro** [comment]
PR in: https://github.com/Youshido/GraphQLBundle/pull/48

**Comment by iainmckay** [comment]
We've run in to this particular recently as well. Could the PR be reviewed?

**Comment by viniychuk** [comment]
Got it in the queue. We will merge it asap.
